### PR TITLE
feat: i18n: add korean language (ko-KR)

### LINKED ISSUE #4332: Add Korean language for Dify
### Self Checks

- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] Pleas do not modify this template :) and fill in all the required fields.

### 1. Is this request related to a challenge you're experiencing?

I'm planning to add Korean language support to Dify!

### 2. Describe the feature you'd like to see

image

Dify pages translated into natural Korean, as shown in the capture above.

### 3. How will this feature improve your workflow or experience?

Koreans who struggle with English will have no major difficulties accessing Dify. :)

### 4. Additional context or comments

_No response_

### 5. Can you help us with this feature?

- [X] I am interested in contributing to this feature.

### PR DESCRIPTION
**Pull request by qkdxorjs1002**
# Description

Add support for i18n Korean language (ko-KR)
First, I used machine translation to translate from Japanese to Korean. 
Then, I manually corrected awkwards and missings.

Fixes #4332

### My Test Result 
image

## Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update, included: [Dify Document](https://github.com/langgenius/dify-docs)
- [ ] Improvement, including but not limited to code refactoring, performance optimization, and UI/UX improvement
- [ ] Dependency upgrade

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] TODO

# Suggested Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I ran `dev/reformat`(backend) and `cd web && npx lint-staged`(frontend) to appease the lint gods
- [ ] `optional` I have made corresponding changes to the documentation 
- [ ] `optional` I have added tests that prove my fix is effective or that my feature works
- [ ] `optional` New and existing unit tests pass locally with my changes

### CODE REVIEW

File: web/i18n/language.ts [hunk 1]
```diff
@@ -74,8 +74,8 @@ export const languages = [
   {
     value: 'ko-KR',
     name: '한국어 (대한민국)',
-    example: '안녕, Dify!',
-    supported: false,
+    example: '안녕하세요, Dify!',
```
Comment (qkdxorjs1002) [inline web/i18n/language.ts:77]: Correction to correspond to Korean honorifics.

File: web/i18n/language.ts [hunk 2]
```diff
@@ -74,8 +74,8 @@ export const languages = [
...
+    supported: true,
```
Comment (qkdxorjs1002) [inline web/i18n/language.ts:78]: I modified the boolean value to 'true' according to the i18n translation contribution guidelines.

File: web/i18n/language.ts [hunk 3]
```diff
@@ -152,7 +152,7 @@ export const NOTICE_I18N = {
     fr_FR: 'Our system will be unavailable from 19:00 to 24:00 UTC on August 28 for an upgrade. For questions, kindly contact our support team (support@dify.ai). We value your patience.',
     de_DE: 'Our system will be unavailable from 19:00 to 24:00 UTC on August 28 for an upgrade. For questions, kindly contact our support team (support@dify.ai). We value your patience.',
     ja_JP: 'Our system will be unavailable from 19:00 to 24:00 UTC on August 28 for an upgrade. For questions, kindly contact our support team (support@dify.ai). We value your patience.',
-    ko_KR: 'Our system will be unavailable from 19:00 to 24:00 UTC on August 28 for an upgrade. For questions, kindly contact our support team (support@dify.ai). We value your patience.',
+    ko_KR: '시스템이 업그레이드를 위해 UTC 시간대로 8월 28일 19:00 ~ 24:00에 사용 불가될 예정입니다. 질문이 있으시면 지원 팀에 연락주세요 (support@dify.ai). 최선을 다해 답변해드리겠습니다.',
```
Comment (qkdxorjs1002) [inline web/i18n/language.ts:155]: I did a natural interpretation when translating the announcement into Korean.

**Review by qkdxorjs1002 — COMMENTED** [review]
It's looking good for me. double check

**Review by crazywoola — APPROVED** [review]

**Comment by crazywoola** [comment]
Thanks for your contribution :) If you have further questions please reach out to on on discord. pinkbanana9815

**Pull request merged**